In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
df = pd.read_csv('ar_properties.csv')
df

id    ad_type  start_date    end_date  \
0       DyVXfkpKygVBKuUk5olH+A==  Propiedad  2020-08-22  2020-09-03   
1       9naojilaMecJN4jlQiTkGg==  Propiedad  2020-08-22  2020-09-04   
2       tlCPRJPjoDEUzuuCelemAQ==  Propiedad  2020-08-22  2020-08-31   
3       Zw3b91glQUO3HNrM5fPYlQ==  Propiedad  2020-08-22  2020-09-04   
4       bsU81gm9JEgtZCbTYgvykg==  Propiedad  2020-08-22  2020-09-04   
...                          ...        ...         ...         ...   
999995  jMYhwydKqUje8HCx7RzwJQ==  Propiedad  2020-12-04  9999-12-31   
999996  FsP7ornzqnK1CvUL8quQ9Q==  Propiedad  2020-12-04  9999-12-31   
999997  fJLVUIziPl31ACoaE/fj1g==  Propiedad  2020-12-04  9999-12-31   
999998  7mW3bUSJFA0RBOj2xtBi+w==  Propiedad  2020-12-04  9999-12-31   
999999  rEynGyGrrsxNeG0ib5vUBA==  Propiedad  2020-12-04  2020-12-21   

        created_on        lat        lon         l1  \
0       2020-08-22 -34.407468 -58.957367  Argentina   
1       2020-08-22 -37.996039 -57.542509  Argentina   
2       2020-08-22 -31.380187 -58.009182  Argentina   
3       2020-08-22 -27.494106 -55.123455  Argentina   
4       2020-08-22 -32.948856 -60.630464  Argentina   
...            ...        ...        ...        ...   
999995  2020-12-04 -32.958863 -60.665633  Argentina   
999996  2020-12-04 -34.596587 -58.376893  Argentina   
999997  2020-12-04 -34.576445 -58.432603  Argentina   
999998  2020-12-04 -32.866636 -68.841144  Argentina   
999999  2020-12-04 -34.447524 -58.631552  Argentina   

                                  l2             l3  ... bathrooms  \
0           Bs.As. G.B.A. Zona Norte          Pilar  ...       NaN   
1       Buenos Aires Costa Atlántica  Mar del Plata  ...       NaN   
2                         Entre Ríos      Concordia  ...       1.0   
3                           Misiones          Oberá  ...       1.0   
4                           Santa Fe        Rosario  ...       1.0   
...                              ...            ...  ...       ...   
999995                      Santa Fe        Rosario  ...       7.0   
999996               Capital Federal         Retiro  ...       5.0   
999997               Capital Federal        Palermo  ...       2.0   
999998                       Mendoza            NaN  ...      20.0   
999999      Bs.As. G.B.A. Zona Norte          Tigre  ...       4.0   

       surface_total  surface_covered     price  currency  price_period  \
0           133139.0              NaN       NaN       NaN       Mensual   
1              687.0            687.0       NaN       NaN       Mensual   
2               80.0             80.0       NaN       NaN       Mensual   
3                NaN              NaN       NaN       NaN       Mensual   
4               76.0             66.0       NaN       NaN       Mensual   
...              ...              ...       ...       ...           ...   
999995         400.0            300.0  400000.0       USD       Mensual   
999996         465.0            465.0    3700.0       USD       Mensual   
999997         615.0            425.0  570000.0       USD       Mensual   
999998         450.0            450.0  800000.0       USD       Mensual   
999999         350.0            300.0    7000.0       USD       Mensual   

                                                    title  \
0       VENTA - Lote Industrial 130.000m2 - Parque Ind...   
1                                    Casa - Santa Cecilia   
2                                        Casa - Concordia   
3                                         Terreno - Obera   
4                                  Departamento - Rosario   
...                                                   ...   
999995  Se vende imponente casa a minutos del centro d...   
999996                        Edificio Comercial - Retiro   
999997  Edificio Comercial  en excelente zona de Paler...   
999998                       Edificio Comercial - Capital   
999999  INCREÍBLE ECASA A LA LAGUNA DESDE DIC a FEBRER...   

                                         

In [3]:
df

id    ad_type  start_date    end_date  \
0       DyVXfkpKygVBKuUk5olH+A==  Propiedad  2020-08-22  2020-09-03   
1       9naojilaMecJN4jlQiTkGg==  Propiedad  2020-08-22  2020-09-04   
2       tlCPRJPjoDEUzuuCelemAQ==  Propiedad  2020-08-22  2020-08-31   
3       Zw3b91glQUO3HNrM5fPYlQ==  Propiedad  2020-08-22  2020-09-04   
4       bsU81gm9JEgtZCbTYgvykg==  Propiedad  2020-08-22  2020-09-04   
...                          ...        ...         ...         ...   
999995  jMYhwydKqUje8HCx7RzwJQ==  Propiedad  2020-12-04  9999-12-31   
999996  FsP7ornzqnK1CvUL8quQ9Q==  Propiedad  2020-12-04  9999-12-31   
999997  fJLVUIziPl31ACoaE/fj1g==  Propiedad  2020-12-04  9999-12-31   
999998  7mW3bUSJFA0RBOj2xtBi+w==  Propiedad  2020-12-04  9999-12-31   
999999  rEynGyGrrsxNeG0ib5vUBA==  Propiedad  2020-12-04  2020-12-21   

        created_on        lat        lon         l1  \
0       2020-08-22 -34.407468 -58.957367  Argentina   
1       2020-08-22 -37.996039 -57.542509  Argentina   
2       2020-08-22 -31.380187 -58.009182  Argentina   
3       2020-08-22 -27.494106 -55.123455  Argentina   
4       2020-08-22 -32.948856 -60.630464  Argentina   
...            ...        ...        ...        ...   
999995  2020-12-04 -32.958863 -60.665633  Argentina   
999996  2020-12-04 -34.596587 -58.376893  Argentina   
999997  2020-12-04 -34.576445 -58.432603  Argentina   
999998  2020-12-04 -32.866636 -68.841144  Argentina   
999999  2020-12-04 -34.447524 -58.631552  Argentina   

                                  l2             l3  ... bathrooms  \
0           Bs.As. G.B.A. Zona Norte          Pilar  ...       NaN   
1       Buenos Aires Costa Atlántica  Mar del Plata  ...       NaN   
2                         Entre Ríos      Concordia  ...       1.0   
3                           Misiones          Oberá  ...       1.0   
4                           Santa Fe        Rosario  ...       1.0   
...                              ...            ...  ...       ...   
999995                      Santa Fe        Rosario  ...       7.0   
999996               Capital Federal         Retiro  ...       5.0   
999997               Capital Federal        Palermo  ...       2.0   
999998                       Mendoza            NaN  ...      20.0   
999999      Bs.As. G.B.A. Zona Norte          Tigre  ...       4.0   

       surface_total  surface_covered     price  currency  price_period  \
0           133139.0              NaN       NaN       NaN       Mensual   
1              687.0            687.0       NaN       NaN       Mensual   
2               80.0             80.0       NaN       NaN       Mensual   
3                NaN              NaN       NaN       NaN       Mensual   
4               76.0             66.0       NaN       NaN       Mensual   
...              ...              ...       ...       ...           ...   
999995         400.0            300.0  400000.0       USD       Mensual   
999996         465.0            465.0    3700.0       USD       Mensual   
999997         615.0            425.0  570000.0       USD       Mensual   
999998         450.0            450.0  800000.0       USD       Mensual   
999999         350.0            300.0    7000.0       USD       Mensual   

                                                    title  \
0       VENTA - Lote Industrial 130.000m2 - Parque Ind...   
1                                    Casa - Santa Cecilia   
2                                        Casa - Concordia   
3                                         Terreno - Obera   
4                                  Departamento - Rosario   
...                                                   ...   
999995  Se vende imponente casa a minutos del centro d...   
999996                        Edificio Comercial - Retiro   
999997  Edificio Comercial  en excelente zona de Paler...   
999998                       Edificio Comercial - Capital   
999999  INCREÍBLE ECASA A LA LAGUNA DESDE DIC a FEBRER...   

                                         

# vemos las variables

In [3]:
df.columns #con la funcion columns se puede ver los nombres de las variables

Index(['id', 'ad_type', 'start_date', 'end_date', 'created_on', 'lat', 'lon',
       'l1', 'l2', 'l3', 'l4', 'l5', 'l6', 'rooms', 'bedrooms', 'bathrooms',
       'surface_total', 'surface_covered', 'price', 'currency', 'price_period',
       'title', 'description', 'property_type', 'operation_type'],
      dtype='object')

# vemos los tipos de datos que estan en la base

In [4]:
df.info()  #con la funcion info() se puede ser los tipos de datos que tenemos en cada variable

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000000 entries, 0 to 999999
Data columns (total 25 columns):
 #   Column           Non-Null Count    Dtype  
---  ------           --------------    -----  
 0   id               1000000 non-null  object 
 1   ad_type          1000000 non-null  object 
 2   start_date       1000000 non-null  object 
 3   end_date         1000000 non-null  object 
 4   created_on       1000000 non-null  object 
 5   lat              894233 non-null   float64
 6   lon              894191 non-null   float64
 7   l1               1000000 non-null  object 
 8   l2               1000000 non-null  object 
 9   l3               965273 non-null   object 
 10  l4               306162 non-null   object 
 11  l5               5530 non-null     object 
 12  l6               0 non-null        float64
 13  rooms            714179 non-null   float64
 14  bedrooms         649933 non-null   float64
 15  bathrooms        765122 non-null   float64
 16  surface_total    47

In [5]:
df.columns

Index(['id', 'ad_type', 'start_date', 'end_date', 'created_on', 'lat', 'lon',
       'l1', 'l2', 'l3', 'l4', 'l5', 'l6', 'rooms', 'bedrooms', 'bathrooms',
       'surface_total', 'surface_covered', 'price', 'currency', 'price_period',
       'title', 'description', 'property_type', 'operation_type'],
      dtype='object')

In [8]:
df2 = df.rename(columns={'ad_type':'tipo_aviso',
                'start_date':'inicio_publicacion',
                'end_date':'cierre_publicacion',
                'created_on':'fecha_alta_publi',
                'l1':'pais_publi',
                'l2':'provincia',
                'l3':'ciudad',
                'l4':'barrio',
                'l5':'sector_propiedad',
                'l6':'tipo_operacion',
                'rooms':'cantidad_habitaciones',
                'bathrooms':'cantidad_banios',
                'surface_total':'superf_total',
                'surface_covered':'superf_cubierta',
                'price':'precio',
                'currency':'tipo_moneda_publicada',
                'price_period':'period_precio_public',
                'title':'titulo_anuncio',
                'description':'descripc_anuncio',
                'development':'desarrollo_inmobiliario',
                'property_type':'descripcion_propiedad',
                'operation_type':'tipo_operacion'})

In [9]:
df2.columns

Index(['id', 'tipo_aviso', 'inicio_publicacion', 'cierre_publicacion',
       'fecha_alta_publi', 'lat', 'lon', 'pais_publi', 'provincia', 'ciudad',
       'barrio', 'sector_propiedad', 'tipo_operacion', 'cantidad_habitaciones',
       'bedrooms', 'cantidad_banios', 'superf_total', 'superf_cubierta',
       'precio', 'tipo_moneda_publicada', 'period_precio_public',
       'titulo_anuncio', 'descripc_anuncio', 'descripcion_propiedad',
       'tipo_operacion'],
      dtype='object')

# Realizamos un analisis preliminar de las variables para saber que contienen 

# Realizamos distribuciones de frecuencia para ver las categorias de estas variables

In [14]:
df2.columns

Index(['id', 'tipo_aviso', 'inicio_publicacion', 'cierre_publicacion',
       'fecha_alta_publi', 'lat', 'lon', 'pais_publi', 'provincia', 'ciudad',
       'barrio', 'sector_propiedad', 'tipo_operacion', 'cantidad_habitaciones',
       'bedrooms', 'cantidad_banios', 'superf_total', 'superf_cubierta',
       'precio', 'tipo_moneda_publicada', 'period_precio_public',
       'titulo_anuncio', 'descripc_anuncio', 'descripcion_propiedad',
       'tipo_operacion'],
      dtype='object')

In [16]:
tipo_aviso = df2[{'tipo_aviso'}]
print(tipo_aviso)

       tipo_aviso
0       Propiedad
1       Propiedad
2       Propiedad
3       Propiedad
4       Propiedad
...           ...
999995  Propiedad
999996  Propiedad
999997  Propiedad
999998  Propiedad
999999  Propiedad

[1000000 rows x 1 columns]


C:\Users\zarac\AppData\Local\Temp\ipykernel_6868\4148198608.py:1: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.
  tipo_aviso = df2[{'tipo_aviso'}]


In [18]:
#realicé una distribución de frecuencias para ver las categorias de la variable

frecuencia_tipo_aviso = (tipo_aviso.groupby("tipo_aviso").
agg(frecuencias=("tipo_aviso","count")))

print(frecuencia_tipo_aviso)                



            frecuencias
tipo_aviso             
Propiedad       1000000


In [19]:
df2.columns

Index(['id', 'tipo_aviso', 'inicio_publicacion', 'cierre_publicacion',
       'fecha_alta_publi', 'lat', 'lon', 'pais_publi', 'provincia', 'ciudad',
       'barrio', 'sector_propiedad', 'tipo_operacion', 'cantidad_habitaciones',
       'bedrooms', 'cantidad_banios', 'superf_total', 'superf_cubierta',
       'precio', 'tipo_moneda_publicada', 'period_precio_public',
       'titulo_anuncio', 'descripc_anuncio', 'descripcion_propiedad',
       'tipo_operacion'],
      dtype='object')

In [23]:
inicio_publicacion= df2[{'inicio_publicacion'}]

print (inicio_publicacion)

       inicio_publicacion
0              2020-08-22
1              2020-08-22
2              2020-08-22
3              2020-08-22
4              2020-08-22
...                   ...
999995         2020-12-04
999996         2020-12-04
999997         2020-12-04
999998         2020-12-04
999999         2020-12-04

[1000000 rows x 1 columns]


C:\Users\zarac\AppData\Local\Temp\ipykernel_6868\1095296772.py:1: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.
  inicio_publicacion= df2[{'inicio_publicacion'}]


In [31]:
frecuencia_inicio_publicacion = (inicio_publicacion.groupby("inicio_publicacion").
agg(frecuencias=("inicio_publicacion","count")))

print(frecuencia_inicio_publicacion)

                    frecuencias
inicio_publicacion             
2020-05-24                  975
2020-05-25                 1204
2020-05-26                 2420
2020-05-27                 2371
2020-05-28                 3046
...                         ...
2021-06-12                 1512
2021-06-13                  589
2021-06-15                 3538
2021-06-16                 3916
2021-06-17                 3879

[362 rows x 1 columns]


In [32]:
df2.columns

Index(['id', 'tipo_aviso', 'inicio_publicacion', 'cierre_publicacion',
       'fecha_alta_publi', 'lat', 'lon', 'pais_publi', 'provincia', 'ciudad',
       'barrio', 'sector_propiedad', 'tipo_operacion', 'cantidad_habitaciones',
       'bedrooms', 'cantidad_banios', 'superf_total', 'superf_cubierta',
       'precio', 'tipo_moneda_publicada', 'period_precio_public',
       'titulo_anuncio', 'descripc_anuncio', 'descripcion_propiedad',
       'tipo_operacion'],
      dtype='object')

In [33]:
provincia = df2[{'provincia'}]

print(provincia)

                           provincia
0           Bs.As. G.B.A. Zona Norte
1       Buenos Aires Costa Atlántica
2                         Entre Ríos
3                           Misiones
4                           Santa Fe
...                              ...
999995                      Santa Fe
999996               Capital Federal
999997               Capital Federal
999998                       Mendoza
999999      Bs.As. G.B.A. Zona Norte

[1000000 rows x 1 columns]


C:\Users\zarac\AppData\Local\Temp\ipykernel_6868\268846037.py:1: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.
  provincia = df2[{'provincia'}]


In [34]:
frecuencia_provincia = (provincia.groupby("provincia").
agg(frecuencias=("provincia","count")))

print(frecuencia_provincia)

                              frecuencias
provincia                                
Bs.As. G.B.A. Zona Norte           184310
Bs.As. G.B.A. Zona Oeste            76786
Bs.As. G.B.A. Zona Sur             131431
Buenos Aires Costa Atlántica        73759
Buenos Aires Interior               23975
California                              4
Canelones                             720
Capital Federal                    265125
Catamarca                             466
Chaco                                1716
Chubut                               1529
Colonia                              1213
Corrientes                           1573
Córdoba                             66274
Entre Ríos                           6750
Florida                               518
Formosa                                30
Jujuy                                 715
La Pampa                             1660
La Rioja                              487
Maldonado                            7784
Maryland                          

In [39]:
df_mask=df2['provincia']=='Capital Federal'

df_mask=df2['Capital Federal']=='Capital Federal'
positions = np.flatnonzero(df_mask)
filtered_df=df2.iloc[positions]
print(filtered_df)

KeyError: 'Capital Federal'

# Tenemos que limpiar las categorias y quedarnos solamente con los 265125 de CABA

In [35]:
df2.columns

Index(['id', 'tipo_aviso', 'inicio_publicacion', 'cierre_publicacion',
       'fecha_alta_publi', 'lat', 'lon', 'pais_publi', 'provincia', 'ciudad',
       'barrio', 'sector_propiedad', 'tipo_operacion', 'cantidad_habitaciones',
       'bedrooms', 'cantidad_banios', 'superf_total', 'superf_cubierta',
       'precio', 'tipo_moneda_publicada', 'period_precio_public',
       'titulo_anuncio', 'descripc_anuncio', 'descripcion_propiedad',
       'tipo_operacion'],
      dtype='object')

In [36]:
ciudad = df2[{'ciudad'}]
print(ciudad)

               ciudad
0               Pilar
1       Mar del Plata
2           Concordia
3               Oberá
4             Rosario
...               ...
999995        Rosario
999996         Retiro
999997        Palermo
999998            NaN
999999          Tigre

[1000000 rows x 1 columns]


C:\Users\zarac\AppData\Local\Temp\ipykernel_6868\2430239774.py:1: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.
  ciudad = df2[{'ciudad'}]


In [38]:
frecuencia_ciudad = (ciudad.groupby("ciudad").
agg(frecuencia=("ciudad","count")))

print(frecuencia_ciudad)

                frecuencia
ciudad                    
9 de Julio              16
Abasto                 851
Acceso Norte             2
Acebal                  11
Achiras                  5
...                    ...
Zárate                 304
orquincó                 2
Álvarez                 30
Ámbul                    6
Ángel Gallardo           7

[1367 rows x 1 columns]


# FRECUENCIAS DE LAS VARIABLES
'description', 'property_type', 'operation_type'

In [6]:
variables = df[{'description', 'property_type', 'operation_type'}]

print(variables)

           operation_type                                        description  \
0                   Venta  Lote (Nro.50) mide 133.139 m2\n<br>Valor: U$S ...   
1                   Venta  Conjunto edilicio de casa y departamentos en b...   
2                Alquiler  Casa de un dormitorio, cocina, baño, living co...   
3                   Venta  IMPORTANTE PROPIEDAD EN VENTA EN OBERA MISIONE...   
4                   Venta  PORTAL DE LOS MARINOS<br>Es un complejo de viv...   
...                   ...                                                ...   
999995              Venta  Esta excelente propiedad se encuentra ubicada ...   
999996           Alquiler  GRAN EDIFICIO PETIT HOTEL ESTILO FRANCÉS  CON ...   
999997              Venta  Edificio Comercial CON ENTRADA INDEPENDIENTE S...   
999998              Venta  Local 450m² con Cloaca en 25 de mayo, Argentin...   
999999  Alquiler temporal  ESPECTACULAR CASA A LA LAGUNA  -BARRIO SANTA B...   

          property_type  
0            

C:\Users\zarac\AppData\Local\Temp\ipykernel_6868\4016162468.py:1: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.
  variables = df[{'description', 'property_type', 'operation_type'}]


In [10]:
frecuencia_operation_type = (df.groupby("operation_type").
agg(frecuencias=("operation_type","count")))

print(frecuencia_operation_type)

                   frecuencias
operation_type                
Alquiler                207490
Alquiler temporal        41679
Venta                   750831


In [11]:
frecuencia_property_type = (df.groupby("property_type").
agg(frecuencias=("property_type","count")))

print(frecuencia_property_type)

                 frecuencias
property_type               
Casa                  245726
Casa de campo           2251
Cochera                13276
Departamento          457907
Depósito                7872
Local comercial        45593
Lote                  121311
Oficina                32390
Otro                   27837
PH                     45837


# Analisis de nulos en la base

In [12]:
# identificamos si tiene o no tiene nulls la variable l4

l4 = None  
if l4 is None:
 print('tiene nulos')
else:
 print('no tiene nulos')

tiene nulos


In [29]:
# al ver que tiene nulls, lo que hacemos es corroborar si el resto de las variables tambien tienen nulls. 

df

print(len(df)-df.count()) #identificamos los nulos sobre el total de las variables de la base

id                       0
ad_type                  0
start_date               0
end_date                 0
created_on               0
lat                 105767
lon                 105809
l1                       0
l2                       0
l3                   34727
l4                  693838
l5                  994470
l6                 1000000
rooms               285821
bedrooms            350067
bathrooms           234878
surface_total       522169
surface_covered     512244
price                41757
currency             44509
price_period        570130
title                    1
description             42
property_type            0
operation_type           0
dtype: int64


# analizamos las categorias de las variables 

In [ ]:
# armar listas para buscar las comunas

str_match = [s for s in my_list if s.__contains__("ack")]
print(str_match)

# PROCEDO A LIMPIAR LAS VARIABLES QUE NO VAMOS A USAR PARA ESTE ANÁLISIS

In [12]:
df2.columns

Index(['id', 'tipo_aviso', 'inicio_publicacion', 'cierre_publicacion',
       'fecha_alta_publi', 'lat', 'lon', 'pais_publi', 'provincia', 'ciudad',
       'barrio', 'sector_propiedad', 'tipo_operacion', 'cantidad_habitaciones',
       'bedrooms', 'cantidad_banios', 'superf_total', 'superf_cubierta',
       'precio', 'tipo_moneda_publicada', 'period_precio_public',
       'titulo_anuncio', 'descripc_anuncio', 'descripcion_propiedad',
       'tipo_operacion'],
      dtype='object')

In [ ]:
df2 = df2.drop(columnas = [
       'cantidad_habitaciones',
       'bedrooms', 'cantidad_banios', 
       'superf_total', 'superf_cubierta',
       'descripc_anuncio'])
df2.head()

In [ ]:
# actualizar el infoerm de acuerdo a los comentarios realizadoe en el dia de la fecha
# ir detectando que elelemtnos puede ser analizados desde el marco conceptual de smart city/ ciudades globales/mercantilizacion del suelo, etc
#



In [1]:
from scipy import stats

class capa():
  def __init__(self, n_neuronas_capa_anterior, n_neuronas, funcion_act):
    self.funcion_act = funcion_act
    self.b  = np.round(stats.truncnorm.rvs(-1, 1, loc=0, scale=1, size= n_neuronas).reshape(1,n_neuronas),3)
    self.W  = np.round(stats.truncnorm.rvs(-1, 1, loc=0, scale=1, size= n_neuronas * n_neuronas_capa_anterior).reshape(n_neuronas_capa_anterior,n_neuronas),3)